src/styles/main.scss

```diff
@@ -1,3 +1,61 @@
+@import "./utils/variables";
+
 body {
   margin: 0;
+  display: flex;
+  height: 100vh;
+  justify-content: center;
+  align-items: center;
+}
+
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  flex-direction: row;
+
+  width: (($daySize + $gapSize) * 7);
+  padding: 10px;
+  gap: $gapSize;
+
+  &__day {
+    display: flex;
+    align-items: center;
+    justify-content: center;
+
+    box-sizing: border-box;
+    width: $daySize;
+    height: $daySize;
+
+    font-family: Arial, Helvetica, sans-serif;
+    font-size: 30px;
+    background-color: #eee;
+    border: 1px solid #000;
+
+    transition: background-color 0.5s, transform 0.5s;
+
+    &:hover {
+      background-color: pink;
+      transform: translateY(-20px);
+      cursor: pointer;
+    }
+
+    @each $day, $coll in $days {
+      .calendar--start-day--#{$day} > &:first-child {
+        margin-left: $coll * ($daySize + $gapSize);
```

Comment:
This selector and the one for month length, can be moved under .calendar instead of here and simplified. You can use `index` to get the index positions without having to store them in the list
```css
@each $day in $days{
  $i: index($day, $days);
  &--start-day--#{$day} :first-child {
    margin-left: ($i - 1) * ($daySize + $gapSize);
```
index starts from 1, not 0 like JS